src/Illuminate/Support/ServiceProvider.php

```diff
@@ -10,6 +10,10 @@
 use Illuminate\Database\Eloquent\Factory as ModelFactory;
 use Illuminate\View\Compilers\BladeCompiler;
 
+/**
+ * @property array<class-string, class-string> $bindings All of the container bindings that should be registered.
+ * @property array<class-string, class-string> $singletons All of the singletons that should be registered.
```

Comment:
These types are wrong, because bindings (and singletons) don't need to be class strings. They should be typehinted as `string` instead.

Reply:
Feel free to send in follow up PR's 👍

Reply:
https://github.com/laravel/framework/pull/52308